provider/datagen/src/lib.rs

```diff
@@ -690,14 +690,13 @@ fn test_keys_from_file() {
 
 #[test]
 fn test_keys_from_bin() {
+    let mut buffer = std::env::var("CARGO_MANIFEST_DIR")
```

Comment:
Given that this is a single file, I'd prefer to `include_bytes!` it and write it to a temp file in order to call the API. This way it's not required to be there at runtime.

Reply:
I don't want to muck with tempfiles in a test when it's already in a file. I'm fine with requiring it at runtime.

Reply:
I think to test a method that accepts `Path` we should use a mock file system. Rust's mock file system is `/tmp`.

In the bigger picture I want to get to a point where any test that uses a single file just `includes!` it. `CARGO_MANIFEST_DIR` and panicking on missing files is not something I want in our tests.

Reply:
> Rust's mock file system is /tmp

This would just break on Windows.

Reply:
I mean `std::env::temp_dir` obviously. Please try to address my actual concern.

Reply:
I didn't know that existed! I'm still iffy on that; since there can be other stuff in the tempdir, and non-Cargo build systems may also not like files getting written to `/tmp` (many such build systems are hermetic for reproducability)

Reply:
"writes to a file during a test" is the kind of thing that causes flakes in my experience

Reply:
If the goal is to use more `include!`s (which I'm very much in favor of) I think the better option is to refactor `keys_from_file()` and `keys_from_bin()` into a pair of functions, one of which just takes a bag of bytes.

Reply:
please do that then